doc/guides/device_mgmt/smp_groups/smp_group_8.rst

```diff
@@ -0,0 +1,213 @@
+    +--------+--------------+----------------+
+
+CBOR payload of successful response command:
+
+.. code-block:: none
+
+    {
+        (str)"off"      : (uint)
+        (str)"data"     : (str)
+        (str)"rc"       : (int)0
+    }
+
+In case of error the payload takes form:
+
+.. code-block:: none
+
+    {
+        (str)"rc"           : (int)
+        (str,opt)"rsn"      : (str)
+    }
+
+where
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "off"                 | offset the response is for                        |
+    +-----------------------+---------------------------------------------------+
+    | "data"                | chunk of data read from file; it is CBOR encoded  |
+    |                       | stream of bytes with embeeded size;               |
+    |                       | "data" appears only in responses where "rc" is 0  |
+    +-----------------------+---------------------------------------------------+
+    | "len"                 | length of file, this field is only mandatory      |
+    |                       | when "off" is 0                                   |
+    +-----------------------+---------------------------------------------------+
+    | "rc"                  | :ref:`mcumgr_smp_protocol_status_codes`           |
+    +-----------------------+---------------------------------------------------+
+    | "rsn"                 | :ref:`mcumgr_smp_protocol_status_codes`           |
+    +-----------------------+---------------------------------------------------+
+
+In case when "rc" is not 0, success, the other fields will not appear.
+
+File upload
+***********
+
+Allows to upload a file to a specified location. Command will automatically overwrite
+existing file or create a new one if it does not exist at specified path.
+The protocol supports stateless upload where each requests carries different chunk
+of a file and it is client side responsibility to track progress of upload.
+
+Mcumgr server side re-opens a file for each subsequent request, and current
+specification does not provide means to identify subsequent requests as
+belonging to specified download session.  This means that the  file is not
+locked in any way or exclusively owned by mcumgr, for the time of download
+session, and may change between requests or even be removed.
+
+.. note::
+    Wiredly the current implementation is half-stateless as is able to hold
+    single upload context, holding  information on ongoing upload, that consists
+    of bool flag indicating in-progress upload, last successfully uploaded offset
+    and total length only.
+
+File upload request
+===================
+
+File upload request header
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``2``  | ``8``        |  ``1``         |
```

Comment:
Command ID looks to be 0 (shared between upload and download): https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c#L219

Reply:
Unresolved

Reply:
Good catch.